### PR TITLE
Use TypeORM entities the correct way

### PR DESCRIPTION
**Pull request by fragosti**
In launch-kit we were not using typeORM decorators because they didn't look good when transpiling to js, but that is no longer a problem.

I also removed `reflect-metadata` as it wasn't being used.

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -1,14 +1,10 @@
 // tslint:disable:custom-no-magic-numbers
 import { BigNumber } from '0x.js';
 import { assert } from '@0x/assert';
-import * as crypto from 'crypto';
```
Comment (BMillman19) [inline src/config.ts]: Looks like this accidentally got committed as a part of this branch

File: src/entity/SignedOrderEntity.ts [hunk 2]
```diff
@@ -1,65 +1,99 @@
...
-            type: 'varchar',
-        },
-        remainingFillableTakerAssetAmount: {
-            type: 'varchar',
-        },
-    },
-});
+import { Column, Entity, PrimaryColumn } from 'typeorm';
+
+@Entity({ name: 'signed_orders' })
```
Comment (BMillman19) [inline src/entity/SignedOrderEntity.ts]: should this be `signed_order`?

File: src/entity/SignedOrderEntity.ts [hunk 3]
```diff
@@ -1,65 +1,99 @@
...
+export class SignedOrderEntity {
+    @PrimaryColumn({ name: 'hash', type: 'varchar' })
+    public hash?: string;
```
Comment (BMillman19) [inline src/entity/SignedOrderEntity.ts]: Are these all optional?

File: src/entity/SignedOrderEntity.ts [hunk 4]
```diff
@@ -1,65 +1,99 @@
...
+        this.expirationTimeSeconds = opts.expirationTimeSeconds;
+        this.makerFee = opts.makerFee;
+        this.takerFee = opts.takerFee;
+        this.makerFeeAssetData = opts.makerFeeAssetData;
+        this.takerFeeAssetData = opts.takerFeeAssetData;
+        this.makerAssetAmount = opts.makerAssetAmount;
+        this.takerAssetAmount = opts.takerAssetAmount;
+        this.salt = opts.salt;
+        this.signature = opts.signature;
+        this.remainingFillableTakerAssetAmount = opts.remainingFillableTakerAssetAmount;
```
Comment (BMillman19) [inline src/entity/SignedOrderEntity.ts]: I wonder if there is any way we don't need to make changes in 3 different places in order to add a new property

File: src/services/orderbook_service.ts [hunk 5]
```diff
@@ -71,16 +71,16 @@ export class OrderBookService {
         quoteAssetData: string,
     ): Promise<OrderbookResponse> {
         const connection = getDBConnection();
-        const bidSignedOrderModels = (await connection.manager.find(SignedOrderModel, {
+        const bidSignedOrderEntitys = (await connection.manager.find(SignedOrderEntity, {
```
Comment (BMillman19) [inline src/services/orderbook_service.ts]: sp: `bidSignedOrderEntities`

Comment (BMillman19) [reply src/services/orderbook_service.ts]: same below

File: src/entity/SignedOrderEntity.ts (hunk 3 above)
Comment (fragosti) [reply src/entity/SignedOrderEntity.ts]: Yes -- I am not sure **why** but I would rather leave them this way for now.

File: src/entity/SignedOrderEntity.ts (hunk 4 above)
Comment (fragosti) [reply src/entity/SignedOrderEntity.ts]: I can't think of a really clean way. FWIW this object existed before I just moved it here.

**Pull request merged**